File: src/main/java/com/example/repository/UserRepository.java
package com.example.repository;

import com.example.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;
import java.util.List;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    List<User> findByDepartment(String department);
    
    @Query("SELECT u FROM User u JOIN FETCH u.activities WHERE u.department = ?1")
    List<User> findByDepartmentWithActivities(String department);
}

File: user-service/src/main/kotlin/com/financial/control/user/repository/UserRepository.kt
package com.financial.control.user.repository

import com.financial.control.user.model.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface UserRepository : JpaRepository<User, Long> {
    
    fun findByEmail(email: String): Optional<User>
    
    fun findByCpf(cpf: String): Optional<User>
    
    fun existsByEmail(email: String): Boolean
    
    fun existsByCpf(cpf: String): Boolean
    
    @Query("SELECT u FROM User u WHERE u.active = true")
    fun findAllActiveUsers(): List<User>
    
    @Query("SELECT u FROM User u WHERE u.id = :id AND u.active = true")
    fun findActiveUserById(id: Long): Optional<User>
}
